src/util/file.cpp

```diff
@@ -171,11 +170,32 @@ read_file(const std::string& path, size_t size_hint)
     // it's actually needed.
     if (has_utf16_le_bom(result)) {
       result.erase(0, 2); // Remove BOM.
-      std::u16string result_as_u16((result.size() / 2) + 1, '\0');
-      result_as_u16 = reinterpret_cast<const char16_t*>(result.c_str());
-      std::wstring_convert<std::codecvt_utf8_utf16<char16_t>, char16_t>
-        converter;
-      result = converter.to_bytes(result_as_u16);
+      if (result.empty())
+        return result;
+
+      std::wstring result_as_u16((result.size() / 2) + 1, '\0');
+      result_as_u16 = reinterpret_cast<const wchar_t*>(result.c_str());
+
+      const int size = WideCharToMultiByte(CP_UTF8,
+                                           0,
+                                           result_as_u16.c_str(),
+                                           int(result_as_u16.size()),
+                                           nullptr,
+                                           0,
+                                           nullptr,
+                                           nullptr);
+      if (size <= 0)
+        return {};
```

Comment:
I suggest returning an error instead of silently returning an empty string. Perhaps something like this?

```suggestion
      if (size <= 0) {
        return nonstd::make_unexpected(
          FMT("Failed to convert {} from UTF-16LE to UTF-8: {}",
              path,
              Win32Util::error_message(size)));
      }
```

(Untested.)

A test case for this in `unittest/test_util_file.cpp` would be appreciated.

Reply:
Done